streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1003,17 +1003,12 @@ private void checkIfUnexpectedUserSpecifiedConsumerConfig(final Map<String, Obje
 
         // verify that producer batch config is no larger than segment size, then add topic configs required for creating topics
         final Map<String, Object> topicProps = originalsWithPrefix(TOPIC_PREFIX, false);
+        final Map<String, Object> producerProps = getClientPropsWithPrefix(PRODUCER_PREFIX, ProducerConfig.configNames());
 
-        if (topicProps.containsKey(topicPrefix(TopicConfig.SEGMENT_INDEX_BYTES_CONFIG))) {
```

Comment:
This is a piggy-backed fix to not use SEGMENT_INDEX_BYTES_CONFIG but SEGMENT_BYTES_CONFIG